signer/src/transaction_signer.rs

```diff
@@ -385,6 +423,31 @@ where
 
         Ok(())
     }
+
+    #[tracing::instrument(skip(self, msg))]
+    async fn send_message(
+        &mut self,
+        msg: impl Into<message::Payload>,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<(), error::Error> {
+        let msg = msg
+            .into()
+            .to_message(
+                bitcoin::BlockHash::from_slice(bitcoin_chain_tip)
+                    .map_err(error::Error::SliceConversion)?,
+            )
+            .sign_ecdsa(&self.signer_private_key)?;
```

Comment:
Hmmm, this shouldn't fail (like, conceptually it shouldn't fail; it definitely can fail given how we've set things up). Not a change for this PR, we can update later.

Reply:
This should be infallible. The result types that are propagated over the interface seem to be FFI artifacts stemming from the lack of type safety from the underlying C library.